build-tools/cmake/xa_macros.cmake

```diff
@@ -32,18 +32,23 @@ macro(xa_common_prepare)
     set(DSO_SYMBOL_VISIBILITY "hidden")
   endif()
 
+  #
+  # Currently not supported by NDK clang, but worth considering when it is eventually supported:
+  #
+  #  -fsanitize=safe-stack
+  #
+
   # Don't put the leading '-' in options
   set(XA_COMPILER_FLAGS
     fno-strict-aliasing
     ffunction-sections
     funswitch-loops
     finline-limit=300
     fvisibility=${DSO_SYMBOL_VISIBILITY}
-    fstack-protector
+    fstack-protector-strong
+    fstrict-return
     Wa,--noexecstack
-    Wformat
```

Comment:
Why move `-Wformat`/etc. into `src/monodroid/CMakeLists.txt`?  Won't that alter `src/sqlite-xamarin/CMakeLists.txt`?  Or do we not care because of PR #3375 and `src/sqlite-xamarin` will disappear?

Reply:
We won't be fixing any 3rd party code, so adding those flags when building it only increases the noise.